src/transformers/models/llama/modeling_llama.py

```diff
@@ -989,7 +992,10 @@ def forward(
         if position_ids is None:
             position_ids = cache_position.unsqueeze(0)
 
-        causal_mask = self._update_causal_mask(attention_mask, inputs_embeds, cache_position, past_seen_tokens)
+        static_cache_max_length = self.layers[0].self_attn.past_key_value.get_max_length() if is_static_cache else None
```

Comment:
not super sure the static_cache_max_length should be extracted here, owuld rather hide everything in the _update_causal_mask. That's the only nit

Reply:
tried to address it in [899fc6b](https://github.com/huggingface/transformers/pull/30348/commits/899fc6bd99220d2445c8cfade8372dc1aa19ae8c)

Reply:
A lot better thanks